src/components/register-form.js

```diff
@@ -12,30 +16,26 @@ export class RegisterForm extends React.Component {
       passwordConfirm: '',
       preferences: [],
       age: 20,
-      isMale: true
+      isMale: true,
+      touched: {
+        email: false,
+        password: false,
+        passwordConfirm: false,
+      }
     }
   }
 
-  onChangeEmail = e => {
-    const { value: email } = e.target
+  handleBlur = (field) => (evt) => {
     this.setState({
-      email
-    })
-  };
-
-  onChangePassword = e => {
-    const { value: password } = e.target
-    this.setState({
-      password
-    })
-  };
+      touched: { ...this.state.touched, [field]: true },
+    });
+  }
 
-  onChangeConfirmPassword = e => {
-    const { value: passwordConfirm } = e.target
+  onChangeInputText = e => {
     this.setState({
-      passwordConfirm
-    })
-  };
+      [e.target.id]: e.target.value,
```

Comment:
you can destructure the target in the function signature, it would make the code more readable.

Reply:
:O it's true